Classes/common/touchdb/TDBlobStore.m

```diff
@@ -187,6 +190,8 @@ - (NSInteger)deleteBlobsExceptWithKeys:(NSSet*)keysToKeep withDatabase:(FMDataba
             [NSData dataWithBytes:oneRow.key.bytes length:sizeof(oneRow.key.bytes)];
         if ([keysToKeep containsObject:curKeyData]) {
             // Do not delete blob. It is an exception.
+            [filesToKeep addObject:oneRow.blobFilename];
+
             continue;
         }
 
```

Comment:
After these changes it seems redundant to attempt to delete the file here, given it would be deleted anyway by `deleteFilesExceptWithFilenames` lower down?

Is the reason for this that otherwise we might delete a row in the attachments table, then assign the same filename later but still have the old file with that file name still on disk (so end up with the wrong attachment data because the new attachment data would not be saved over it?)